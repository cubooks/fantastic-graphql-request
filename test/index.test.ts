import SchemaFetchRequest from '../src/index';

test('generate default type schema', () => {
    const schema = new SchemaFetchRequest();

    const result = schema.generateRequest({
        action: 'getUserInfo',
        responseFields: ['id', 'email'],
    });

    expect(result).toMatchSnapshot();
});

test('generate custom type schema', () => {
    const schema = new SchemaFetchRequest({
        type: 'mutation Mutation',
    });

    const result = schema.generateRequest({
        action: 'registerUser',
        variables: [{ name: 'data', type: 'UserAttributesType' }],
        responseFields: ['id'],
        data: { data: [] },
    });

    expect(result).toMatchSnapshot();
});

test('generate three response fields schema', () => {
    const schema = new SchemaFetchRequest({
        type: 'mutation Mutation',
    });

    const result = schema.generateRequest({
        action: 'updateUserProfile',
        variables: [{ name: 'data', type: 'UserProfileAttributesType' }],
        responseFields: [
            'profileWrited',
            [
                'profile',
                [
                    'firstName',
                    'lastName',
                    'nickName',
                    'gender',
                    'photoId',
                    'wins',
                    'losses',
                    'draws',
                    'weight',
                    'city',
                    'tapologyLink',
                    'sherdogLink',
                ],
            ],
        ],
        data: { data: [] },
    });

    expect(result).toMatchSnapshot();
});