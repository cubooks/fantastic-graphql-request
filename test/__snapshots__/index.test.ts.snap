// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate custom type schema 1`] = `
Object {
  "body": "{\\"query\\":\\"mutation Mutation($data: UserAttributesType){registerUser(data: $data){id}}\\",\\"variables\\":{\\"data\\":[]}}",
  "credentials": "include",
  "headers": Object {
    "Accept": "application/json",
    "Content-Type": "application/json",
  },
  "method": "post",
}
`;

exports[`generate default type schema 1`] = `
Object {
  "body": "{\\"query\\":\\"query Query{getUserInfo{id,email}}\\",\\"variables\\":{}}",
  "credentials": "include",
  "headers": Object {
    "Accept": "application/json",
    "Content-Type": "application/json",
  },
  "method": "post",
}
`;

exports[`generate three response fields schema 1`] = `
Object {
  "body": "{\\"query\\":\\"mutation Mutation($data: UserProfileAttributesType){updateUserProfile(data: $data){profileWrited,profile{firstName,lastName,nickName,gender,photoId,wins,losses,draws,weight,city,tapologyLink,sherdogLink}}}\\",\\"variables\\":{\\"data\\":[]}}",
  "credentials": "include",
  "headers": Object {
    "Accept": "application/json",
    "Content-Type": "application/json",
  },
  "method": "post",
}
`;
